yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java

```diff
@@ -310,6 +320,10 @@ protected DumperOptions buildDumperOptions(int jsonFeatures, int yamlFeatures,
         if (Feature.USE_PLATFORM_LINE_BREAKS.enabledIn(_formatFeatures)) {
             opt.setLineBreak(DumperOptions.LineBreak.getPlatformLineBreak());
         }
+
+        if (Feature.USE_LONG_KEYS.enabledIn(_formatFeatures)) {
+            opt.setMaxSimpleKeyLength(1024);
```

Comment:
This is the setting that would need to be encapsulated in a separate class, with a method that gets `opt` passed in, sets the maximum length. Since that class is only loaded when method called (first "active use") it is possible
to avoid failure when `YAMLGenerator` class itself is initialized.
Class loading exception should trigger throwing of exception that suggests too old SnakeYAML dependency.